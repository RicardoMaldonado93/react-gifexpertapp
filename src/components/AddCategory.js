import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) =>setInputValue(target.value);

    const handleSubmit = (e)=>{ 
        e.preventDefault(); 

        if(inputValue.trim().length > 2){
            setCategory( items =>[  inputValue, ...items ]);
            setInputValue("");
            // document.querySelector("input").placeholder = "ingrese categoria..."
        }

    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}

