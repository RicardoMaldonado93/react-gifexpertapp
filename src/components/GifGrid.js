import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    const { data:images, loading } = useFetchGif( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> ~ { category } ~ </h3>
            { loading && <p className="animate__animated animate__flash">cargando...</p> }
            <div className="card-grid animate__animated animate__fadeIn">
                {         
                    images.map( img =>
                    <GifGridItem 
                        {...img}
                        key={img.id}
                    />  
                    )
                }
            </div>
        </>
    )
}
