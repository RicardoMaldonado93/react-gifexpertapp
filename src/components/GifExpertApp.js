import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] })=>{
    
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategory={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category =>
                        <GifGrid 
                            category={category}
                            key={category}
                        />
                    )
                }

            </ol>
            
        </>
    );
}

export default GifExpertApp;