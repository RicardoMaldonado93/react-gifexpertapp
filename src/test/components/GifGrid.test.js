import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGif";

jest.mock("../../hooks/useFetchGif");

describe("Unit test for '<GifGrid>'", () => {
    let wrapper;

    beforeAll(()=>{
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const category = "One Punch";
        wrapper = shallow(<GifGrid category={category}/>);
    })
   
    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();    
    });

    test("show view items with load useFetchGifs",()=>{
        const gifs = [
            {
                id:"ABC",
                title:"one punch",
                url:"https://localhost:8080/prueba"
            }
        ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading:false
        });

        const category = "One Punch";
        wrapper = shallow(<GifGrid category={category}/>);

        expect( wrapper.find("p").exists() ).toBe(false);
        expect( wrapper.find("GifGridItem").length ).toBe( gifs.length );
    });
    
})
