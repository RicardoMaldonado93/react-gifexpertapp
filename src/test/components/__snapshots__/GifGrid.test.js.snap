// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for '<GifGrid>' should render correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     ~ 
    One Punch
     ~ 
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    cargando...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
